LibraryManagement.Data.Tests/DataRequestTests/ExampleTests/GenresTest.cs

```diff
@@ -0,0 +1,109 @@
+﻿using LibraryManagement.Data.DataRequestObjects.Genres;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using LibraryManagement.Data.DataRequestObjects.Examples;
+using LibraryManagement.Data.Tests.Helpers;
+using LibraryManagement.Data.Abstraction;
+using Microsoft.Data.SqlClient;
+using System.ComponentModel.DataAnnotations;
+using LibraryManagement.Domain.Constants;
+
+
+namespace LibraryManagement.Data.Tests.DataRequestTests.ExampleTests
+{
+    public class GenresTest : DataTest
+    {
+        [Fact]
+        public async Task InsertGenre_GivenCorrect_ShouldInsert_Successfully()
+        {
+            var request = new InsertGenres("Philosophy");
+
+            var result = await _dataAccess.ExecuteAsync(request);
+
+            Assert.True(result == 1);
+
+        }
+
+        [Fact]
+        public async Task InsertGenre_Given_GenreName_AlreadyExists_ShouldReturn_NoRowsUpdated()
+        {
+            var genreName = "Fiction";
+            await _dataAccess.ExecuteAsync(new InsertGenres(genreName));
+
+            var rowsAffected = await _dataAccess.ExecuteAsync(new InsertGenres(genreName));
+
+            Assert.Equal(1, rowsAffected);
+        }
+
+        [Fact]
+        public async Task InsertGenre_Given_NameIsNull_ShouldThrow_SqlException()
+        {
+            var request = new InsertGenres(null!);
+
+            await Assert.ThrowsAsync<SqlException>(async () => await _dataAccess.ExecuteAsync(request));
+        }
+
+        [Fact]
+        public async Task InsertGenre_Given_NameIsEmpty_ShouldThrow_SqlException()
+        {
+            var request = new InsertGenres("");
+
+            await Assert.ThrowsAsync<SqlException>(async () => await _dataAccess.ExecuteAsync(request));
+        }
+
+        [Fact]
+        public async Task InsertGenre_Given_NameIsWhitespace_ShouldThrow_SqlException()
+        {
+            var request = new InsertGenres("   ");
```

Comment:
I love that you set up tests for NULL, Empty, and Whitespace. Great to make sure you know how to expect the application to behave in these scenarios. We can probably fit these all into one Test method though, using a 'Theory' instead of a 'Fact'.

It'd look something like this:
```
[Theory]
[InlineData(null!)]
[InlineData("")]
[InlineData("  ")]
public async Task InsertGenre_Given_NameIsInvalid_ShouldThrow_SqlException(string invalidName)
{
   var request = new InsertGenre(invalidName);
   
   await Assert.ThrowsAsync<SqlException>(async () => await _dataAccess.ExecuteAsync(request));
}

```